javascript/0540-single-element-in-a-sorted-array.js

```diff
@@ -0,0 +1,20 @@
+// Time Complexity: O(log n)
+// Space Complexity: O(1)
+
+/**
+ * @param {number[]} nums
+ * @return {number}
+ */
+var singleNonDuplicate = function (nums) {
+    let left = 0,
+        right = nums.length - 2;
+
+    while (left <= right) {
+        const mid = (left + right) >> 1;
+
+        if (nums[mid] === nums[mid ^ 1]) left = mid + 1;
```

Comment:
Suggestion: This is a more advanced binary search. You are using mid and mid + 1. This small tweak can used in m `bitonic arrays`. Better to define the comparison to indicate that. 

```js
const mid1 = ((left + right) >> 1);
const mid2 = (mid ^ 1);
```